src/org/parkerici/image_utils/ij/hyperstack.clj

```diff
@@ -0,0 +1,35 @@
+(ns org.parkerici.image-utils.ij.hyperstack
+  (:import [net.imglib2 Cursor RandomAccess]
+           [net.imglib2.view Views])
+  (:require [clojure.string :as str]))
+
+; Takes a hyperstack image and splits it into its individual slices.
+; Each slice is an ItervalView object that must be converted back into an image before being saved.
+; Returns a map of the format {"dimension_string" IntervalView}, where dimension string is of the format
+; 0_1_2 where the first each position represents a dimension and the value there represents the slice number. 
+(defn img->slices
+  ([img] (img->slices img ""))
+  ([img dimensions]
+  ; If we are down to 2 dimensions (width and height) then stop recurring and return the image
+  ; Otherwise split on the 3rd dimension and recur on the split images.
+   (if  (= (.numDimensions img) 2)
+     {dimensions img}
+     (apply merge (for [x (range (.dimension img 2))]
```

Comment:
Good catch, but after thinking about it I think that lazy generation is good for this function. Depending on the size of image being split, it might be more memory efficient to wait until the slices are being consumed to actually generate them.